Modules/pyexpat.c

```diff
@@ -1816,25 +1739,45 @@ MODULE_INITFUNC(void)
 
     if (PyModule_AddStringConstant(errors_module, "__doc__",
                                    "Constants used to describe "
-                                   "error conditions.") < 0)
-        return NULL;
+                                   "error conditions.") < 0) {
+        goto error;
+    }
 
-    if (PyModule_AddObject(errors_module, "codes", codes_dict) < 0)
-        return NULL;
-    if (PyModule_AddObject(errors_module, "messages", rev_codes_dict) < 0)
-        return NULL;
+    if (PyModule_AddObject(errors_module, "codes", codes_dict) < 0) {
```

Comment:
Hum. Reference counting is hard. Py_XDECREF(codes_dict) on goto error is wrong.

I suggest to call Py_INCREF(codes_dict) before, and add Py_DECREF(codecs_dict) before return 0. Same for rev_codes_dict.

Reply:
@vstinner 

I can change this block to:
```
    Py_INCREF(codes_dict);
    if (PyModule_AddObject(errors_module, "codes", codes_dict) < 0) {
        Py_DECREF(codes_dict);
        goto error;
    }

    Py_INCREF(rev_codes_dict);
    if (PyModule_AddObject(errors_module, "messages", rev_codes_dict) < 0) {
        Py_DECREF(rev_codes_dict);
        goto error;
    }

    Py_DECREF(codes_dict);
    Py_DECREF(rev_codes_dict);
    return 0;
```

But what about the goto error which can happen from MYCONST?  I think I need to decref these dicts in those cases via the goto error label.

Reply:
Thanks, the updated code is more regular and so simpler to review!